torch/nn/functional.py

```diff
@@ -547,6 +547,22 @@ def nll_loss(input, target, weight=None, size_average=True):
     return f(input, target)
 
 
+def poisson_nll_loss(log_input, target, size_average=True):
+    r"""Poisson negative log likelihood loss.
+
+    Drops :math:`log(target!)` term. See :class:`~torch.nn.PoissonNLLLoss` for details.
+
+    Args:
+        log_input: :math:`log(input)`
+        target: :math:`target \sim Pois(input)`
+    """
+    loss = torch.exp(log_input) - target * log_input
```

Comment:
I could do it, but this seems to have very restricted usage. The approximation requires evaluation of `log(target)`, but `target` for poissonian processes often happens to be zero. 
By the way probably we should add two losses? One is ordinary `PoissonNLLLoss` (without `log(input)`) and the other is `LogPoissonNLLLoss` which is current implementation